python_modules/dagster/dagster/_core/definitions/freshness_checks/sensor.py

```diff
@@ -130,69 +61,62 @@ def build_sensor_for_freshness_checks(
         default_status=default_status,
     )
     def the_sensor(context: SensorEvaluationContext) -> Optional[RunRequest]:
-        cursor = (
-            deserialize_value(context.cursor, FreshnessCheckSensorCursor)
-            if context.cursor
-            else FreshnessCheckSensorCursor.empty()
+        left_off_asset_check_key = (
+            AssetCheckKey.from_user_string(context.cursor) if context.cursor else None
         )
-        current_timestamp = pendulum.now("UTC").timestamp()
-        check_keys_to_run = []
-        for asset_check in freshness_checks:
-            for asset_check_spec in asset_check.check_specs:
-                prev_evaluation_timestamp = cursor.get_last_evaluation_timestamp(
-                    asset_check_spec.key
-                )
-                if should_run_again(asset_check_spec, prev_evaluation_timestamp, current_timestamp):
-                    check_keys_to_run.append(asset_check_spec.key)
-        context.update_cursor(
-            serialize_value(cursor.with_updated_evaluations(check_keys_to_run, current_timestamp))
+        start_time = pendulum.now("UTC")
+        checks_iter = ordered_iterator_freshness_checks_starting_with_key(
+            left_off_asset_check_key, freshness_checks
         )
-        if check_keys_to_run:
-            return RunRequest(asset_check_keys=check_keys_to_run)
+        check_key = next(checks_iter, None)
+        checks_to_evaluate = []
+        while (
```

Comment:
I think worth emitting a log message in the case that we skip checks

Reply:
Added verbose logging for all cases.